### PR TITLE
Update pipeline to increase gstreamer performance

### PR DESCRIPTION
**Pull request by joaofigueiredobisect**

### CODE REVIEW

File: cpp/libs/ossrf_gstreamer_api/lib/src/receiver/st2110_20_receiver_plugin.cpp [hunk 1]
```diff
@@ -41,32 +41,36 @@ struct gst_st2110_20_receiver_impl : gst_receiver_plugin_t
...
             "application/x-rtp, media=(string)video, clock-rate=(int)90000, encoding-name=(string)RAW, "
             "sampling=(string)RGB, width=(string)640, height=(string)480");
         g_object_set(G_OBJECT(source), "caps", caps, NULL);
 
-        // Add pipeline rtpjitterbuffer
-        auto* jitter_buffer = gst_element_factory_make("rtpjitterbuffer", NULL);
-        BST_ENFORCE(jitter_buffer != nullptr, "Failed creating GStreamer element jitter_buffer");
-        BST_ENFORCE(gst_bin_add(GST_BIN(pipeline), jitter_buffer), "Failed adding jitter_buffer to the pipeline");
-
         // Add pipeline queue1
```
Comment (pedro-alves-ferreira) [inline cpp/libs/ossrf_gstreamer_api/lib/src/receiver/st2110_20_receiver_plugin.cpp:57]: I don't think it's a good idea to remove the jitter buffer.

File: cpp/libs/ossrf_gstreamer_api/lib/src/receiver/st2110_20_receiver_plugin.cpp [hunk 2]
```diff
@@ -41,32 +41,36 @@ struct gst_st2110_20_receiver_impl : gst_receiver_plugin_t
...
         auto* queue1 = gst_element_factory_make("queue", NULL);
         BST_ENFORCE(queue1 != nullptr, "Failed creating GStreamer element queue");
         BST_ENFORCE(gst_bin_add(GST_BIN(pipeline), queue1), "Failed adding queue to the pipeline");
+        g_object_set(queue1, "max-size-buffers", 3, NULL);
```
Comment (pedro-alves-ferreira) [inline cpp/libs/ossrf_gstreamer_api/lib/src/receiver/st2110_20_receiver_plugin.cpp:61]: `queue1` handles RTP buffers, while `queue2` handles full frames. Each frame may have several thousands of packets, so I don't think it is a good idea to use the same value (3) for both.
Probably the best solution is to work based on time, for example, you could set both to 200 ms.
Also, make sure you set `max-size-time`, `max-size-buffers` and `max-size-bytes`. They all have defaults and we probably don't want to rely on them. If we go with my suggestion (using `max-size-time`), then we should set both `max-size-buffers` and `max-size-bytes` to `0`, in order to disable those checks and rely only on time.

Comment (pedro-alves-ferreira) [inline cpp/libs/ossrf_gstreamer_api/lib/src/receiver/st2110_20_receiver_plugin.cpp:61]: Don't hardcode constants in a function call (3 in this case). Always define `constexpr` values with meaningful names and explain why you chose that value.

**Review by pedro-alves-ferreira — CHANGES_REQUESTED** [review]

**Review by pedro-alves-ferreira — APPROVED** [review]
LGTM

**Review by massis08 — APPROVED** [review]
LGTM

**Pull request merged**